cm2350/e200_gdb.py

```diff
@@ -132,15 +145,9 @@ def _postClientAttach(self, addr):
         # halt.
 
         logger.info("Client attached: %r", repr(addr))
-        logger.info("Halting processor")
-
-        self._halt_reason = signal.SIGTRAP
-        self.emu.halt_exec()
```

Comment:
Does attaching not halt the processor?

Reply:
Or is this functionality handled by Vivisect's GDBSTUB/Server code?

Reply:
This took a little while for me to reason through. In this case the `handleInterrupts` is called by the e200z7 core when a DebugException occurs so it just needs to wait on the event that allows execution to continue. When given the `-g` flag at start the mpc5674 class calls the gdbstub's `waitForClient` function during init which queues up a debug exception that causes the processor to wait until the gdb client says "go". Which is meant to mimic how qemu starts up.

But your comment made me think... when GDB attaches to a running process it's supposed to halt, right? And I just checked and that is not what happens. It's probably unusual that someone might want to attach to the target without specifying the `-g` flag? but it is currently possible. I may need to reconsider how the target halts.

Reply:
I suppose, with -g (QEMU only, btw), the target is already stopped.
I prefer this functionality better, actually... the ability to attach without halting.  I just thought it was the other, more annoying way.

Reply:
I prefer it also... except that attaching while it's running doesn't actually halt the process. so I should revisit that

Reply:
this is fixed now. the emulator properly halts when the debugger attaches if it's already running